collector/perfos.go

```diff
@@ -0,0 +1,132 @@
+// returns data points from Win32_PerfRawData_PerfOS_System class (undocumented)
+
+package collector
+
+import (
+	"log"
+
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	Factories["perfos"] = NewPerfOSCollector
+}
+
+// A PerfOSCollector is a Prometheus collector for WMI metrics
+type PerfOSCollector struct {
+	ContextSwitchesPersec     *prometheus.Desc
+	ExceptionDispatchesPersec *prometheus.Desc
+	ProcessorQueueLength      *prometheus.Desc
+	SystemCallsPersec         *prometheus.Desc
+	SystemUpTime              *prometheus.Desc
+	Threads                   *prometheus.Desc
+}
+
+// NewPerfOSCollector ...
+func NewPerfOSCollector() (Collector, error) {
+	const subsystem = "perfos"
+
+	return &PerfOSCollector{
+		ContextSwitchesPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "context_switches_per_sec"),
+			"PerfOS_System.ContextSwitchesPersec",
+			nil,
+			nil,
+		),
+		ExceptionDispatchesPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "exception_dispatches_per_sec"),
+			"PerfOS_System.ExceptionDispatchesPersec",
+			nil,
+			nil,
+		),
+		ProcessorQueueLength: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "processor_queue_length"),
+			"PerfOS_System.ProcessorQueueLength",
+			nil,
+			nil,
+		),
+		SystemCallsPersec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "system_calls_per_sec"),
+			"PerfOS_System.SystemCallsPersec",
+			nil,
+			nil,
+		),
+		SystemUpTime: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "system_up_time"),
+			"PerfOS_System.SystemUpTime",
+			nil,
+			nil,
+		),
+		Threads: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "threads"),
+			"PerfOS_System.Threads",
+			nil,
+			nil,
+		),
+	}, nil
+}
+
+// Collect sends the metric values for each metric
+// to the provided prometheus Metric channel.
+func (c *PerfOSCollector) Collect(ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ch); err != nil {
+		log.Println("[ERROR] failed collecting os metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+type Win32_PerfRawData_PerfOS_System struct {
+	ContextSwitchesPersec     uint32
```

Comment:
I think this is actually a total. This is the current value on my machine:
`ContextSwitchesPersec          : 666198955`